consumer/pdata/generated_common.go

```diff
@@ -185,3 +185,22 @@ func (es AnyValueArray) MoveAndAppendTo(dest AnyValueArray) {
 	}
 	*es.orig = nil
 }
+
+// Filter calls f sequentially for each element present in the slice.
+// If f returns true, filter deletes the given element from the slice.
```

Comment:
I think this is inverse of what most filter implementations do. At least in Python and Java where true keeps the element:

```
In [6]: list(filter(lambda x: x == 3, [1,2,3]))
Out[6]: [3]
```

Reply:
Changed name to `RemoveAll`, as per your suggestion filter usually returns a new slice not doing in place.